### PR TITLE
[gas] consider memory leaked for values entering the event natives

### PR DESCRIPTION
**Pull request by vgao1996**
By default, the memory usage tracker considers the memory released when values enter a native function. This is however not the desired behavior for the event natives as the values stay in the event buffer until the end of the transaction. This gets it fixed by considering the memory leaked conceptually.

The change is gated by a new gas feature version (4).

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5377)

### CODE REVIEW

**Review by wrwg — COMMENTED** [review]
The TODO sounds perhaps stronger than it needs to be -- can we make this more neutral in  the code and perhaps point to an issue? Folks are scanning through our codes, so comments like this can be misinterpreted.

**Comment by vgao1996** [comment]
@wrwg done. See https://github.com/aptos-labs/aptos-core/issues/5485

**Review by wrwg — APPROVED** [review]

File: aptos-move/aptos-gas/src/gas_meter.rs [hunk 1]
```diff
@@ -27,6 +27,8 @@ use move_vm_types::{
 use std::collections::BTreeMap;
 
 // Change log:
+// - V4
+//   - Consider memory leaked for event natives
```
Comment (msmouse) [inline aptos-move/aptos-gas/src/gas_meter.rs:31]: @alinush  it'll be nice if you can add your change at v4 as well

**Review by msmouse — APPROVED** [review]

File: aptos-move/aptos-gas/src/gas_meter.rs (hunk 1 above)
Comment (alinush) [reply aptos-move/aptos-gas/src/gas_meter.rs:31]: Ehh, which change?

**Pull request merged**